keyboards/keyboardio/atreus/keymaps/kkokdae/keymap.c

```diff
@@ -0,0 +1,127 @@
+      KC_A,    KC_R,    KC_S,    KC_T,    KC_G,                      KC_M,    KC_N,    KC_E,    KC_I,    KC_O,
+      KC_Z,    KC_X,    KC_C,    KC_D,    KC_V,    _______, _______, KC_K,    KC_H,    KC_COMM, KC_DOT,  KC_ENT,
+			OSMCMD,  OSMOPT,  OSMCTL,  OSMSFT,  FUNC,    KC_TAB,  KC_ESC,  NUMPAD,  SYMBOL,  OSMCTL,  OSMOPT,  OSMCMD
+      ),
+
+  [_QWERTY] = LAYOUT(
+      KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                      KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,
+      KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                      KC_H,    KC_J,    KC_K,    KC_L,    KC_QUOT,
+      KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    _______, _______, KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_ENT,
+			OSMCMD,  OSMOPT,  OSMCTL,  OSMSFT,  FUNC,    KC_TAB,  KC_ESC,  NUMPAD,  SYMBOL,  OSMCTL,  OSMOPT,  OSMCMD
+      ),
+
+  [_FUNC] = LAYOUT(
+      KC_MRWD, KC_VOLU, KC_VOLD, KC_MFFD, KC_MPLY,                   KC_INS,  KC_F7,   KC_F8,   KC_F9,   KC_CAPS,
+      KC_LEFT, KC_UP,   KC_DOWN, KC_RGHT, KC_BRIU,                   KC_F12,  KC_F4,   KC_F5,   KC_F6,   TD(TdL),
+      KC_HOME, KC_PGUP, KC_PGDN, KC_END,  KC_BRID, _______, _______, KC_F10,  KC_F1,   KC_F2,   KC_F3,   KC_F11,
+      _______, _______, _______, KC_ASTG, _______, _______, _______, DNUMPAD, _______, _______, _______, _______
+      ),
+
+  [_NUMPAD] = LAYOUT(
+      _______, _______, _______, _______, _______,                   S(KC_E), KC_7,    KC_8,    KC_9,    S(KC_F),
+      TD(TdL), KC_EQL,  KC_ASTR, KC_PLUS, _______,                   S(KC_C), KC_4,    KC_5,    KC_6,    S(KC_D),
+      TD(TdH), KC_DOT,  KC_SLSH, KC_MINS, _______, _______, _______, S(KC_A), KC_1,    KC_2,    KC_3,    S(KC_B),
+      _______, _______, _______, KC_ASTG, _______, _______, _______, _______, KC_0,    _______, _______, _______
+      ),
+
+  [_SYMBOL] = LAYOUT(
+      KC_LCBR, KC_LBRC, KC_RBRC, KC_RCBR, KC_CIRC,                   KC_DLR,  KC_BSLS, KC_PERC, KC_AT,   KC_DQUO,
+      KC_QUES, KC_EXLM, KC_HASH, KC_ASTR, KC_AMPR,                   KC_PIPE, KC_MINS, KC_PLUS, KC_UNDS, KC_EQL,
+      KC_LT,   KC_LPRN, KC_RPRN, KC_GT,   KC_GRV,  _______, _______, KC_TILD, KC_SLSH, C(KC_B), KC_COLN, KC_SCLN,
+      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+      ),
+};
+
+bool get_custom_auto_shifted_key(uint16_t keycode, keyrecord_t *record) {
+	switch(keycode) {
+		case KC_MINS:
+		case KC_UNDS:
+		case KC_BSPC:
+			return false;
+	}
+
+	switch(keycode & 0xFF) {
+		case KC_1 ... KC_UP:
+			autoshift_disable();
+		default:
+			return false;
+	}
+}
+
+void dance_hex(qk_tap_dance_state_t *state, void *user_data) {
+  switch(state->count) {
+    case 1:
+			SEND_STRING("0x");
+      break;
+    case 2:
+			SEND_STRING("\\x");
+      break;
+  }
+}
+
+void dance_lang(qk_tap_dance_state_t *state, void *user_data) {
+  uint32_t default_layer;
```

Comment:
Since this will be a number between 0-32, at most, a uint8_t is a better idea. 
```suggestion
    uint8_t default_layer;
```